x/evm/keeper/migrate.go

```diff
@@ -49,6 +61,102 @@ func removeExternalTokenBurnerCode(ctx sdk.Context, ck chainKeeper) error {
 	return nil
 }
 
+func migrateCommandBatchSignature(ctx sdk.Context, ck chainKeeper, signer types.Signer, multisig types.MultisigKeeper) error {
+	commandBatchID := ck.getLatestSignedCommandBatchID(ctx)
+	for {
+		commandBatchMetadata := ck.getCommandBatchMetadata(ctx, commandBatchID)
+		if commandBatchMetadata.ID == nil {
+			break
+		}
+		commandBatchID = commandBatchMetadata.PrevBatchedCommandsID
+
+		if commandBatchMetadata.Status != types.BatchSigned {
+			if commandBatchMetadata.Status == types.BatchSigning {
+				setCommandBatchAborted(ctx, ck, commandBatchMetadata)
+			}
+			continue
+		}
+
+		key, ok := multisig.GetKey(ctx, commandBatchMetadata.KeyID)
+		if !ok {
+			break
+		}
+
+		sigID := hex.EncodeToString(commandBatchMetadata.ID)
+		multisigInfo, found := signer.GetMultisigSignInfo(ctx, sigID)
+		signInfo, ok := multisigInfo.(*tsstypes.MultisigInfo)
+		if !found || !ok {
+			return fmt.Errorf("sign info %s not found", sigID)
+		}
+
+		payloadHash := commandBatchMetadata.SigHash.Bytes()
+
+		// convert to multisig
+		newMultisig := multisigTypes.MultiSig{
+			KeyID:       commandBatchMetadata.KeyID,
+			PayloadHash: payloadHash,
+			Sigs:        make(map[string]multisigTypes.Signature),
+		}
+
+		for _, info := range signInfo.Infos {
+			if len(info.Data) < 1 {
+				return fmt.Errorf("validator %s is participant without signature for signing %s", info.Participant.String(), sigID)
```

Comment:
should this be an error?

Reply:
I think so, means the data is corrupted. The validator participated in signing but does not have any signature